clang/lib/Driver/Driver.cpp

```diff
@@ -5202,6 +5263,18 @@ class OffloadingActionBuilder final {
       addInputs(sycl_device_wrapper_libs);
       if (isSpirvAOT || TC->getTriple().isNVPTX())
         addInputs(sycl_device_fallback_libs);
+
+      bool isAOT;
+      bool useNativeBfloatLib = useNativeBfloat(TC, isAOT);
+      if (isAOT &&
+          TC->getTriple().getSubArch() != llvm::Triple::SPIRSubArch_fpga) {
+        // Add native or fallback bfloat16 library.
+        if (useNativeBfloatLib)
+          addInputs(sycl_device_bfloat16_native_lib);
+        else
+          addInputs(sycl_device_bfloat16_fallback_lib);
+      }
```

Comment:
Some kind of driver test is needed to verify the proper library is added for the various AOT targets and devices associated.

Reply:
Tests sycl-bfloat16-lib.cpp and sycl-bfloat16-lib-win.cpp have been added.